react-components/src/hooks/use3dScenes.tsx

```diff
@@ -96,11 +58,99 @@ export const use3dScenes = (
   );
 };
 
-function createModelFromEdge(
-  newModelId: number,
-  newModelRevisionId: number,
-  properties: Cdf3dRevisionProperties
-): AddReveal3DModelOptions {
+function createMapOfScenes(
+  scenes: Array<NodeItem<Record<string, unknown>>> | Array<EdgeItem<Record<string, unknown>>>
+): Record<Space, Record<ExternalId, SceneData>> {
+  return scenes.reduce(
+    (acc, scene) => {
+      const { space, externalId } = scene;
+      const properties = Object.values(
+        Object.values(scene.properties)[0] as Record<string, unknown>
+      )[0] as SceneConfigurationProperties;
+      if (acc[space] === undefined) {
+        acc[space] = {};
+      }
+      if (acc[space][externalId] === undefined) {
+        acc[space][externalId] = {
+          name: properties.name,
+          cadModelOptions: [],
+          image360CollectionOptions: []
+        };
+      }
+
+      return acc;
+    },
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    {} as Record<Space, Record<ExternalId, SceneData>>
+  );
+}
+
+function populateSceneMapWithModels(
+  scene360Images:
+    | Array<NodeItem<Record<string, unknown>>>
+    | Array<EdgeItem<Record<string, unknown>>>,
```

Comment:
Same here, can we be more specific here and get rid of the `as`-es? :sweat_smile:

Reply:
No, unfortunately not. Changed it to NodeItem[] | EdgeItem[] in [808e42b](https://github.com/cognitedata/reveal/pull/4131/commits/808e42b35074ad506aac0d3f47f0e32ad69050fa)